internal/node/reverse_proxy/proxy.go

```diff
@@ -2,59 +2,28 @@ package reverse_proxy
 
 import (
 	"fmt"
-	"io/fs"
-	"net"
-	"net/http"
-	"net/http/httputil"
-	"net/url"
-	"path"
-	"path/filepath"
-	"strings"
-	"time"
 
 	"github.com/eagraf/habitat-new/core/state/node"
-	"github.com/rs/zerolog/log"
 )
 
 type RuleSet struct {
-	rules        map[string]RuleHandler
+	rules        map[string]*node.ReverseProxyRule
 	baseFilePath string // Optional, if set, all file server rules will be relative to this path
 }
 
-func (rs RuleSet) Add(name string, rule RuleHandler) error {
-	if _, ok := rs.rules[name]; ok {
-		return fmt.Errorf("rule name %s is already taken", name)
-	}
-	rs.rules[name] = rule
-	return nil
-}
-
 // AddRule is a wrapper around Add for finding the correct rule handler type.
 func (rs RuleSet) AddRule(rule *node.ReverseProxyRule) error {
-	if rule.Type == ProxyRuleRedirect {
-		url, err := url.Parse(rule.Target)
-		if err != nil {
-			return err
-		}
-		err = rs.Add(rule.ID, &RedirectRule{
-			Matcher:         rule.Matcher,
-			ForwardLocation: url,
-		})
-		if err != nil {
-			return err
-		}
-	} else if rule.Type == ProxyRuleFileServer {
-		err := rs.Add(rule.ID, &FileServerRule{
-			Matcher:  rule.Matcher,
-			Path:     rule.Target,
-			BasePath: rs.baseFilePath,
-		})
-		if err != nil {
-			return err
-		}
-	} else {
-		return fmt.Errorf("unknown rule type %s", rule.Type)
+	if _, ok := rs.rules[rule.ID]; ok {
+		return fmt.Errorf("reverse proxy rule with id %s is already present", rule.ID)
+	}
+
+	// Make sure the rule type is valid.
+	if rule.Type != node.ProxyRuleRedirect && rule.Type != node.ProxyRuleFileServer && rule.Type != node.ProxyRuleEmbeddedFrontend {
```

Comment:
I think if we type the rules properly we won't need this

Reply:
I tried this, but TIL go will infer the type here